emperor/support_files/js/model.js

```diff
@@ -79,52 +91,60 @@ function($, _) {
-   * available metadta.
-   * @property {Array} [plottable] array of plottable objects.
-   * @property {Object} [dimensionRanges] object with two properties "min" and
-   * "max", these determine the ranges over which all the samples span.
+   * @param {string} name A string containing the abbreviated name of the
+   * ordination method.
+   * @param {string[]} ids An array of strings where each string is a sample
+   * identifier
+   * @param {float[]} coords A 2D Array of floats where each row contains the
+   * coordinates of a sample. The rows are in ids order.
+   * @param {float[]} pct_var An Array of floats where each position contains
+   * the percentage explained by that axis
+   * @param {float[]} md_headers An Array of string where each string is a
+   * metadata column header
+   * @param {string[]} metadata A 2D Array of strings where each row contains
+   * the metadata values for a given sample. The rows are in ids order. The
+   * columns are in `md_headers` order.
+   *
+   * @throws {Error} In any of the following cases:
+   * - The number of coordinates does not match the number of samples.
+   * - If there's a coordinate in `coords` that doesn't have the same length as
+   *   the rest.
+   * - The number of samples is different than the rows provided as metadata.
+   * - Not all metadata rows have the same number of fields.
+   *
+   * @return {DecompositionModel}
+   * @constructs DecompositionModel
    *
    */
-
-  /**
-   * @name DecompositionModel
-   *
-   * @class Models all the ordination method data to be plotted
-   *
-   * @param {name} a string containing the abbreviated name of the ordination
-   * method
-   * @param {ids} an Array of strings where each string is a sample identifier
-   * @param {coords} a 2D Array of floats where each row contains the
-   * coordinates of a sample. The rows are in ids order.
-   * @param {pct_var} an Array of floats where each position contains the
-   * percentage explained by that axis
-   * @param {md_headers} an Array of string where each string is a metadata
-   * column header
-   * @param {metadata} a 2D Array of strings where each row contains the
-   * metadata values for a given sample. The rows are in ids order. The columns
-   * are in md_headers order.
-   *
-  **/
   function DecompositionModel(name, ids, coords, pct_var, md_headers, metadata) {
     var num_coords;
+    /**
+     * Abbreviated name of the ordination method used to create the data.
+     * @type {string}
+     */
     this.abbreviatedName = name;
+    /**
+     * List of sample name identifiers.
+     * @type {string[]}
+     */
     this.ids = ids;
+    /**
+     * Percentage explained by each of the axes in the ordination.
+     * @type {float[]}
+     */
     this.percExpl = pct_var;
+    /**
+     * Column names for the metadata in the samples.
+     * @type {string[]}
+     */
     this.md_headers = md_headers;
 
     /*
```

Comment:
This should be fine right, this is not intended to be a jsdoc comment.

On (May-25-16|22:52), Joshua Shorenstein wrote:
> >      this.percExpl = pct_var;
> > +    /**
> > +     * Column names for the metadata in the samples.
> > +     * @type {string[]}
> > +     */
> >      this.md_headers = md_headers;
> >  
> >      /*
> 
> missing a star
> 
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/emperor/pull/491/files/3cafd9d51df45436d540ece101e52e8f12f5074a#r64694564